cmd/object-api-common.go

```diff
@@ -309,3 +310,186 @@ func cleanupDir(storage StorageAPI, volume, dirPath string) error {
 	err := delFunc(retainSlash(pathJoin(dirPath)))
 	return err
 }
+
+func checkRange(offsetBegin, offsetEnd int64, resourceSize int64) (startOffset, length int64, err error) {
```

Comment:
I think you could keep `type httpRange struct`  and have `newHTTPRange()` would do this and you could use `getLength()` and `String()` instead of scattering range logic.